pkg/gateway/transaction.go

```diff
@@ -117,3 +123,81 @@ func (txn *Transaction) Submit(args ...string) ([]byte, error) {
 
 	return response.Payload, nil
 }
+
+// RegisterCommitEvent registers for a commit event for this transaction.
+//  Returns:
+//  the channel that is used to receive the event. The channel is closed after the event is queued.
+func (txn *Transaction) RegisterCommitEvent() <-chan *fab.TxStatusEvent {
+	txn.eventch = make(chan *fab.TxStatusEvent, 100)
```

Comment:
I would try to avoid hardcoding a channel size.

Reply:
Also, it's not a good idea to close a channel outside of parent function. During maintenance devs could lose it and then channel and goroutines will be hanging.

Reply:
Since this just forwards a single commit event for the transaction before closing the channel, this only needs to be size 1.  I've changed it to 1 - I don't think there is a need to override this.